pkg/lib/resourcescope/commands.go

```diff
@@ -105,3 +105,118 @@ func (c *Commands) UpdateScope(ctx context.Context, options *UpdateScopeOptions)
 func (c *Commands) DeleteScope(ctx context.Context, resourceURI string, scope string) error {
 	return c.Store.DeleteScope(ctx, resourceURI, scope)
 }
+
+func (c *Commands) AddScopesToClientID(ctx context.Context, resourceURI, clientID string, scopes []string) ([]*model.Scope, error) {
+	if _, found := c.OAuthConfig.GetClient(clientID); !found {
+		return nil, ErrClientNotFound
+	}
+	resource, err := c.Store.GetResourceByURI(ctx, resourceURI)
+	if err != nil {
+		return nil, err
+	}
+	for _, scopeStr := range scopes {
+		scope, err := c.Store.GetScopeByResourceIDAndScope(ctx, resource.ID, scopeStr)
+		if err != nil {
+			return nil, err
+		}
+		err = c.Store.AddScopeToClientID(ctx, resource.ID, scope.ID, clientID)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
There are N queries here. We should use 1 query to fetch all given scopes, and 1 query to insert them.